orderer/consensus/etcdraft/util.go

```diff
@@ -196,6 +196,43 @@ func ConsensusMetadataFromConfigBlock(block *common.Block) (*etcdraft.ConfigMeta
 	return MetadataFromConfigUpdate(configUpdate)
 }
 
+// ConsensusMetadataFromConfigBlockConfig reads consensus metadata from the configuration block's Config
+func ConsensusMetadataFromConfigBlockConfig(block *common.Block) (*etcdraft.ConfigMetadata, error) {
+	if block == nil {
+		return nil, errors.New("nil block")
+	}
+
+	if !protoutil.IsConfigBlock(block) {
+		return nil, errors.New("not a config block")
+	}
+
+	envelope, err := ConfigEnvelopeFromBlock(block)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot read config")
+	}
+
+	payload, err := protoutil.UnmarshalPayload(envelope.Payload)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to extract payload from envelope")
+	}
+
+	configEnvelope, err := configtx.UnmarshalConfigEnvelope(payload.Data)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to extract config envelope from payload")
+	}
+
+	if configEnvelope.Config != nil && configEnvelope.Config.ChannelGroup != nil {
```

Comment:
It's a style preference, but it looks weird to me now to see things nested like this (and all the fall-out errors after the expected path). Any reason to not invert the logic to return out early for the errors to keep it clear at a quick glance which error goes with which path of the code? 

`	if configEnvelope.Config == nil && configEnvelope.Config.ChannelGroup == nil {
		return nil, errors.New("config block missing channel group")
	}

	ordererConfigGroup, ok := configEnvelope.Config.ChannelGroup.Groups["Orderer"]
	if !ok {
		return nil, errors.New("config block missing orderer group")
	}

	val, ok := ordererConfigGroup.Values["ConsensusType"]
	if !ok {
		return nil, errors.New("config block missing consensus type value")
	}

	return MetadataFromConfigValue(val)
`

Reply:
Good catch, I had originally written this logic while I was still figuring out which portions to check for which ended up resulting in the nested checks. I meant to change this afterwards but it slipped through after I had forgotten about it. Thanks!